trucks.js

```diff
@@ -509,3 +511,65 @@ var foodTrucks = [
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
 // filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
 // filterByFoodType(foodType) - return trucks with associated 'foodType'
+
+
+// getTrucks() - return all trucks
+//function getTrucks(day) {
+//	var sortedTrucks = _.filter(foodTrucks, function (item) {
+//
+//		if (typeof item.schedule === 'undefined') {
+//			return false;
+//		} else {
+//			return (item.schedule.indexOf(day) !== -1);
+//		}
+//	});
+//
+//	return sortedTrucks;
+//}
+function getTrucks() {
+	return foodTrucks;
+}
+
+
+// getTruck(name) - return the truck object matching 'name'
+function getTruck(name) {
+
+	for(var i = 0, l = foodTrucks.length; i < l; i++) {
+
+		// Loop through food trucks until you find the matching name. When you find the name, stop.
+		if (foodTrucks[i].name.toLowerCase() === name.toLowerCase()) {
```

Comment:
If you're getting an error here, I'm guessing it's happening when the name parameter isn't assigned a value or is assigned "undefined'". To make your code more robust, check the value you're trying to call a method on before calling it if there is a possibility it hasn't been defined: `if (name && foodTrucks[i].name.toLowerCase() === name.toLowerCase())`  Here the if statement will stop evaluating and return false if 'name' is undefined, and there will be no attempt to call toLowerCase() on it which could result in an error.